packages/x-charts/src/ChartsTooltip/ChartsItemTooltipContent.tsx

```diff
@@ -31,14 +32,26 @@ function ChartsItemTooltipContent(props: ChartsItemTooltipContentProps) {
     return null;
   }
   const { color, label, formattedValue, markType } = tooltipData;
+  const itemId =
+    typeof tooltipData.value === 'object' && tooltipData.value !== null && 'id' in tooltipData.value
+      ? tooltipData.value.id
+      : undefined;
```

Comment:
Not a fan of this, but I can't see another way of doing this. 

IMO, it makes sense to provide the item id instead of the data index for the custom mark, but we have no easy way to access the item id in this component, especially because different charts have different data structures. 

Open to suggestions!

Reply:
Migrated to `dataIndex`

Reply:
Did you consider the `tooltipData.identifier`. It's an object which is made to uniquemy target a series item